### PR TITLE
clientv3: use Endpoints(), fix context creation

### PR DESCRIPTION
**Pull request by gyuho**
If overwritten, the previous context should be canceled first.

Replace https://github.com/etcd-io/etcd/pull/10813.

### CODE REVIEW

**Comment by gyuho** [comment]
@spzala can you take a look

File: clientv3/client.go [hunk 1]
```diff
@@ -519,13 +519,17 @@ func (c *Client) roundRobinQuorumBackoff(waitBetween time.Duration, jitterFracti
...
 func (c *Client) checkVersion() (err error) {
 	var wg sync.WaitGroup
-	errc := make(chan error, len(c.cfg.Endpoints))
+
+	eps := c.Endpoints()
+	errc := make(chan error, len(eps))
 	ctx, cancel := context.WithCancel(c.ctx)
 	if c.cfg.DialTimeout > 0 {
-		ctx, cancel = context.WithTimeout(ctx, c.cfg.DialTimeout)
+		cancel()
```
Comment (jingyih) [inline clientv3/client.go]: I believe the behavior is the same with and without this part of the change?

**Comment by spzala** [comment]
@gyuho looking at it in a min, but glad @jingyih already reviewed :-) Sorry, I was helping with an internal event and just got back!

File: clientv3/client.go (hunk 1 above)
Comment (gyuho) [reply clientv3/client.go]: Yes, it doesn't change any behavior, I believe. Just trying to be clear about what we are doing, overwriting the previous cancel function.

Comment (spzala) [reply clientv3/client.go]: Agree that we can go with or without cancel() here.

**Pull request merged**